kafka-rest/src/test/java/io/confluent/kafkarest/controllers/ClusterConfigManagerImplTest.java

```diff
@@ -0,0 +1,366 @@
+  }
+
+  @Test
+  public void getClusterConfig_existingConfig_returnsConfig() throws Exception {
+    expect(clusterManager.getCluster(CLUSTER_ID)).andReturn(completedFuture(Optional.of(CLUSTER)));
+    expect(adminClient.describeConfigs(
+        eq(singletonList(new ConfigResource(ConfigResource.Type.BROKER, ""))),
+        anyObject(DescribeConfigsOptions.class)))
+        .andReturn(describeConfigsResult);
+    expect(describeConfigsResult.values())
+        .andReturn(
+            singletonMap(
+                new ConfigResource(ConfigResource.Type.BROKER, ""),
+                KafkaFuture.completedFuture(CONFIG)));
+    replay(adminClient, clusterManager, describeConfigsResult);
+
+    ClusterConfig config =
+        clusterConfigManager.getClusterConfig(
+            CLUSTER_ID, ClusterConfig.Type.BROKER, CONFIG_1.getName())
+            .get()
+            .get();
+
+    assertEquals(CONFIG_1, config);
+  }
+
+  @Test
+  public void getClusterConfig_nonExistingConfig_returnsEmpty() throws Exception {
+    expect(clusterManager.getCluster(CLUSTER_ID)).andReturn(completedFuture(Optional.of(CLUSTER)));
+    expect(
+        adminClient.describeConfigs(
+            eq(singletonList(
+                new ConfigResource(ConfigResource.Type.BROKER, ""))),
+            anyObject(DescribeConfigsOptions.class)))
+        .andReturn(describeConfigsResult);
+    expect(describeConfigsResult.values())
+        .andReturn(
+            singletonMap(
+                new ConfigResource(ConfigResource.Type.BROKER, ""),
+                KafkaFuture.completedFuture(CONFIG)));
+    replay(adminClient, clusterManager, describeConfigsResult);
+
+    Optional<ClusterConfig> config =
+        clusterConfigManager.getClusterConfig(CLUSTER_ID, ClusterConfig.Type.BROKER, "foobar")
+            .get();
+
+    assertFalse(config.isPresent());
+  }
+
+  @Test
+  public void getClusterConfig_nonExistingCluster_throwsNotFound() throws Exception {
+    expect(clusterManager.getCluster(CLUSTER_ID)).andReturn(completedFuture(Optional.of(CLUSTER)));
+    expect(
+        adminClient.describeConfigs(
+            eq(singletonList(
+                new ConfigResource(ConfigResource.Type.BROKER, ""))),
+            anyObject(DescribeConfigsOptions.class)))
+        .andReturn(describeConfigsResult);
+    expect(describeConfigsResult.values())
+        .andReturn(
+            singletonMap(
+                new ConfigResource(ConfigResource.Type.BROKER, ""),
+                KafkaFutures.failedFuture(new NotFoundException("Cluster not found."))));
+    replay(clusterManager, adminClient, describeConfigsResult);
+
+    try {
+      clusterConfigManager.getClusterConfig(
+          CLUSTER_ID, ClusterConfig.Type.BROKER, CONFIG_1.getName()).get();
+      fail();
+    } catch (ExecutionException e) {
+      assertEquals(NotFoundException.class, e.getCause().getClass());
+    }
+  }
+
+  @Test
+  public void updateClusterConfig_existingConfig_updatesConfigs() throws Exception {
```

Comment:
Shall we refer to config updates by any one of `updateClusterConfig` or `upsertClusterConfig` at all places for consistency ?

Reply:
Acknowledged.

Solved on https://github.com/confluentinc/kafka-rest/pull/686 (want to avoid the merge conflict).